log/sequencer.go

```diff
@@ -230,12 +237,57 @@ func (s Sequencer) initMerkleTreeFromStorage(ctx context.Context, currentRoot tr
 	return s.buildMerkleTreeFromStorageAtRoot(ctx, currentRoot, tx)
 }
 
-// SequenceBatch wraps up all the operations needed to take a batch of queued leaves
-// and integrate them into the tree.
-// TODO(Martin2112): Can possibly improve by deferring a function that attempts to rollback,
-// which will fail if the tx was committed. Should only do this if we can hide the details of
-// the underlying storage transactions and it doesn't create other problems.
-func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, guardWindow, maxRootDurationInterval time.Duration) (int, error) {
+// sequencingTask is used by Sequencer to extract (pre-)sequenced LogLeaf
+// entries from the storage and commit to the ordering.
+type sequencingTask interface {
+	// fetch returns a batch of sequenced entries obtained from storage. If the
+	// storage does not provide ordering, the latter is defined on the fly.
```

Comment:
I didn't really follow the second sentence -- can the caller of this method assume the provided leaves are ordered or not?   For an `interface` in particular, it's good to be clear about the contract that the caller/implementor are agreeing on -- maybe " ... sequenced entries obtained from storage; the returned leaves will have consecutive LeafIndex values starting from the current tree size" or somesuch.